libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -157,11 +157,15 @@ func (n *bridgeNetwork) setupIPTables(ipVersion iptables.IPVersion, maskedAddr *
 			return setupInternalNetworkRules(config.BridgeName, maskedAddr, config.EnableICC, false)
 		})
 	} else {
-		if err = setupIPTablesInternal(config.HostIP, config.BridgeName, maskedAddr, config.EnableICC, config.EnableIPMasquerade, hairpinMode, true); err != nil {
+		hostIP := config.HostIP
+		if ipVersion != iptables.IPv4 {
+			hostIP = nil
+		}
+		if err = setupIPTablesInternal(hostIP, config.BridgeName, maskedAddr, config.EnableICC, config.EnableIPMasquerade, hairpinMode, true); err != nil {
```

Comment:
I see `setipIPTablesInternal` already has some handling for IPv4 <--> IPv6. I haven't looked closely, but wondering the same for this function, and if it should either

- get passed `config` (as a whole)
- or check if `hostIP.To4() == nil` (as it does for other addresses)

https://github.com/moby/moby/blob/6ce5aa1cd5a46e02ffaa3fe0c794642b35601676/libnetwork/drivers/bridge/setup_ip_tables_linux.go#L205-L230

Reply:
I think it would be cleaner to pass `config` as a whole, but I wanted this commit to be the minimal needed to fix the bug  (and add a regression test).  I'll open a separate PR to do some refactoring for code health.